tests/integration/targets/sysrc/main/main.yml

```diff
@@ -0,0 +1,223 @@
+
+    - name: get file content
+      shell: "cat /etc/rc.conf | egrep -v ^\\#"
+      register: sysrc_content1
+
+    - name: Remove from /etc/rc.conf
+      sysrc:
+        name: sysrc_test
+        state: absent
+      register: sysrc_test1_change
+
+    - name: validate results for key removal
+      assert:
+        that:
+          - sysrc_test1 is changed
+          - sysrc_test1_change is not changed
+          - '"sysrc_test1" not in sysrc_content1.stdout_lines'
+          - '"sysrc_test11" not in sysrc_content1.stdout_lines'
+
+    ##
+    ## sysrc - append/subtract
+    ##
+    - name: append value (creates)
+      sysrc:
+        name: sysrc_test_append
+        state: value_present
+        value: test_a
+      register: sysrc_test2_create
+
+    - name: append value (append)
+      sysrc:
+        name: sysrc_test_append
+        state: value_present
+        value: test_b
+      register: sysrc_test2
+
+    - name: get file content
+      shell: "cat /etc/rc.conf | egrep -v ^\\#"
+      register: sysrc_content2
+
+    - name: append same value
+      sysrc:
+        name: sysrc_test_append
+        state: value_present
+        value: test_b
+      register: sysrc_test2_change
+
+    - name: validate results for key appending
+      assert:
+        that:
+          - sysrc_test2_create is changed
+          - sysrc_test2 is changed
+          - sysrc_test2_change is not changed
+          - '"sysrc_test_append=\"test_a test_b\"" in sysrc_content2.stdout_lines'
+
+    - name: Subtract value
+      sysrc:
+        name: sysrc_test_append
+        state: value_absent
+        value: test_b
+      register: sysrc_test3
+
+    - name: get file content
+      shell: "cat /etc/rc.conf | egrep -v ^\\#"
+      register: sysrc_content3
+
+    - name: Subtract value
+      sysrc:
+        name: sysrc_test_append
+        state: value_absent
+        value: test_b
+      register: sysrc_test3_change
+
+    - name: validate results for key subtraction
+      assert:
+        that:
+          - sysrc_test3 is changed
+          - sysrc_test3_change is not changed
+          - '"sysrc_test_append=\"test_a\"" in sysrc_content3.stdout_lines'
+
+    ##
+    ## sysrc - append/subtract with delimiter
+    ##
+    - name: append value (delimiter)
+      sysrc:
+        name: sysrc_test_append
+        state: value_present
+        delim: ','
+        value: test_b
+      register: sysrc_test4
+
+    - name: get file content
+      shell: "cat /etc/rc.conf | egrep -v ^\\#"
+      register: sysrc_content4
+
+    - name: append value again (delimiter)
+      sysrc:
+        name: sysrc_test_append
+        state: value_present
+        delim: ','
+        value: test_b
+      register: sysrc_test4_change
+
+    - name: validate results for key appending
+      assert:
+        that:
+          - sysrc_test4 is changed
+          - sysrc_test4_change is not changed
+          - '"sysrc_test_append=\"test_a,test_b\"" in sysrc_content4.stdout_lines'
```

Comment:
In general I would advice not to create test cases that depend on previous/other test cases. I would start always with a fresh rc.conf file and make the test cases self-contained.

Imagine you refactor the tests and reshuffle them and suddenly they will fail even though you haven't modified the SUT.